vaadin-grid-flow-parent/vaadin-grid-flow/src/main/java/com/vaadin/flow/component/grid/Grid.java

```diff
@@ -3205,13 +3205,51 @@ public void setMultiSort(boolean multiSort) {
      * @see MultiSortPriority
      */
     public void setMultiSort(boolean multiSort, MultiSortPriority priority) {
-        Objects.requireNonNull(priority,
-                "Multi-sort priority must not be null");
         setMultiSort(multiSort);
         updateMultiSortPriority(priority);
     }
 
+    /**
+     * Sets whether multiple column sorting is enabled on the client-side.
+     *
+     * @param multiSort
+     *            {@code true} to enable sorting of multiple columns on the
+     *            client-side, {@code false} to disable
+     * @param onShiftClickOnly
+     *            {@code true} to enable multi-sort by shift-clicking (when
+     *            {@code multiSort = true}), {@code false} for normal multi-sort
+     *            behavior
+     */
+    public void setMultiSort(boolean multiSort, boolean onShiftClickOnly) {
+        getElement().setProperty("multiSortOnShiftClick",
```

Comment:
The following case still has issues:
- Set multi-sort true with onShiftClickOnly
- Set multi-sort false (without using onShiftClickOnly APIs)

The result is that the Grid is still multi-sortable with shift-click:

```java
grid.setMultiSort(true, true);
grid.setMultiSort(false);

Assert.assertEquals("false", grid.getElement().getProperty("multiSortOnShiftClick"));
```

Reply:
Changed to account for this case.